kv/iter.go

```diff
@@ -26,3 +28,58 @@ func NextUntil(it Iterator, fn FnKeyCmp) error {
 	}
 	return nil
 }
+
+// SliceIter is used to iterate slice
+type SliceIter struct {
+	data []*Entry
+	cur  int
+}
+
+// NewSliceIter creates a new SliceIter
+func NewSliceIter(data []*Entry) *SliceIter {
+	return &SliceIter{
+		data,
+		0,
+	}
+}
+
+// GetSlice returns the inner slice
+func (i *SliceIter) GetSlice() []*Entry {
+	return i.data
+}
+
+// Valid returns true if the current iterator is valid.
+func (i *SliceIter) Valid() bool {
+	return i.cur >= 0 && i.cur < len(i.data)
+}
+
+// Key returns the current key.
+func (i *SliceIter) Key() Key {
+	if !i.Valid() {
+		return nil
+	}
+	return i.data[i.cur].Key
+}
+
+// Value returns the current value.
+func (i *SliceIter) Value() []byte {
+	if !i.Valid() {
+		return nil
+	}
+	return i.data[i.cur].Value
+}
+
+// Next goes the next position. Always return error for this iterator
+func (i *SliceIter) Next() error {
+	if !i.Valid() {
+		return errors.New("iterator is invalid")
+	}
+
+	i.cur++
+	return nil
+}
+
+// Close closes the iterator.
+func (i *SliceIter) Close() {
+	i.cur = -1
+}
```

Comment:
If it's not used, why add it?
Any exposed unused code become a maintain burden, because you don't know will it be used by outside package (or maybe tidb-tools) and change it later will break compatibility.